lib/Doctrine/ODM/MongoDB/Persisters/DocumentPersister.php

```diff
@@ -1144,6 +1149,10 @@ private function convertToDatabaseValue(string $fieldName, $value)
             );
         }
 
+        if ($value instanceof BackedEnum && $mapping['enumType']) {
```

Comment:
Given `$mapping['enumType']` can be either `null` or a backed enum's FQCN (guaranteed by metadata validation) we should be OK to do following check (maybe backed by an `\assert` call to satisfy SA) which should be cheaper

```suggestion
        if (isset($mapping['enumType'])) {
```

Reply:
Also I'm not sure about the call being here instead of after next if. Actually I started wondering if we shouldn't ensure that `enumType` can be combined with `type=int|string` (or not set which allows autoconfiguration to kick in). Right now

```
/** @ODM\Field(type="collection", enumType=Suit::class) */
private array $enums = [];
```

looks OK from user perspective but I believe it'll blow up?

Reply:
This `$value instanceof BackedEnum` is necessary because the value might be the enum's backed value too:
```php
$this->dm->createQueryBuilder(Card::class)
            ->field('suit')->equals(Suit::Spades->value)
            ->field('nullableSuit')->equals('C');
```

I'll add a test for that.

Regarding mixing `enumType` with arrays, it will probably blow up if you map it this way. Not sure how could we solve it here though.

Reply:
> Not sure how could we solve it here though.

Not necessarily here, I'll hack  a PR later today to throw a mapping exception if `enumType` is combined with something else than `int` or `string` types. Then we'll have a clear "it's not working message" for userland instead of weird blow ups under the hood.